server/bleep/src/repo/iterator/fs.rs

```diff
@@ -39,37 +39,32 @@ impl FileSource for FileWalker {
         self.file_list.len()
     }
 
-    fn for_each(self, iterator: impl Fn(RepoFile) + Sync + Send) {
+    fn for_each(self, iterator: impl Fn(RepoDirEntry) + Sync + Send) {
         use rayon::prelude::*;
         self.file_list
             .into_par_iter()
             .filter_map(|entry_disk_path| {
-                let buffer = if entry_disk_path.is_file() {
-                    match std::fs::read_to_string(&entry_disk_path) {
+                if entry_disk_path.is_file() {
+                    let buffer = match std::fs::read_to_string(&entry_disk_path) {
                         Err(err) => {
                             warn!(%err, ?entry_disk_path, "read failed; skipping");
                             return None;
                         }
                         Ok(buffer) => buffer,
-                    }
+                    };
+                    Some(RepoDirEntry::File(RepoFile {
+                        buffer,
+                        path: entry_disk_path.to_string_lossy().to_string(),
+                        branches: vec!["head".into()],
```

Comment:
can we make this `"head"` into a static in `super`?

`static HEAD: &str = "HEAD";`